### PR TITLE
DG-25524 - Added centralized role management to sdk

### PR DESCRIPTION
**Pull request by glitchinfinity**

### CODE REVIEW

File: lib/sfrest/centralized_role_management.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module SFRest
+  # We disable that because we need to keep this naming due to connection.rb way of autoloading stuff and rubocop
```
Comment (Meptl) [inline lib/sfrest/centralized_role_management.rb:4]:
```suggestion
  # We need to keep this naming due to the way  connection.rb autoloads things.
```

Comment (Meptl) [reply lib/sfrest/centralized_role_management.rb:4]: Can probably leave out the "and rubocop doesn't like it". We can assume that given we're disabling rubocop

File: lib/sfrest/centralized_role_management.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  # doesn't like it.
+  # rubocop: disable Naming/ClassAndModuleCamelCase
+  # Find Staging envs and stage a set of sites
```
Comment (Meptl) [inline lib/sfrest/centralized_role_management.rb:7]: Remove this line? Or update it?

File: lib/sfrest/centralized_role_management.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    # Get current centralized role management settings
+    #
+    # @return [Array] an array of roles with their associated settings.
+    def centralized_role_mapping
+      @conn.get('/api/v1/centralized-role-management')
+    end
+
+    # Set the centralized role management settings for a given role.
+    # @param [Boolean] enable the enablement of the mapping.
+    # @param [string] factory_role the site factory role.
```
Comment (Meptl) [inline lib/sfrest/centralized_role_management.rb:25]: _Technically_ all the type names are capitalized (and classes)
```suggestion
    # @param [String] factory_role the site factory role.
```
But the codebase is already inconsistent here so /shrug

File: lib/sfrest/centralized_role_management.rb (hunk 1 above)
Comment (glitchinfinity) [reply lib/sfrest/centralized_role_management.rb]: Fixing :D

File: lib/sfrest/centralized_role_management.rb (hunk 2 above)
Comment (glitchinfinity) [reply lib/sfrest/centralized_role_management.rb]: Fixing :D

File: lib/sfrest/centralized_role_management.rb (hunk 3 above)
Comment (glitchinfinity) [reply lib/sfrest/centralized_role_management.rb]: Fixing :D

**Pull request merged**